projects/centre_console/inc/charging_manager.h

```diff
@@ -11,6 +12,6 @@ typedef enum {
   NUM_CHARGING_STATES
 } ChargingState;
 
-StatusCode init_charging_manager(void);
+StatusCode init_charging_manager(const DriveState *drive_state);
 
 ChargingState *get_global_charging_state(void);
```

Comment:
This was here before, but why is this by a pointer instead of just the charging state ...

Reply:
Good question, at this point it's just for matching convention with the other centre console modules. May as well be consistent?

Reply:
idk, exporting internal state pointers is fine (although not ideal) if used in the test context. Since this is a function that other modules actually use, I feel like they shouldn't be exporting access to the internal state if you don't expect them to change it

Reply:
Fair enough. Fixed, involved some refactoring in main event generator.